File: app/src/main/java/com/example/globoplay_desafio_mobile/viewmodels/watchlist/WatchListViewModel.java
package com.example.globoplay_desafio_mobile.viewmodels.watchlist;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.example.globoplay_desafio_mobile.models.movies.MovieModel;
import com.example.globoplay_desafio_mobile.repositories.WatchListRepository;
import java.util.List;

public class WatchListViewModel extends ViewModel {

    private WatchListRepository watchListRepository;

    //Constructor
    public WatchListViewModel(){
        watchListRepository = WatchListRepository.getInstance();
    }

    //Getter favorite movies
    public MutableLiveData<List<MovieModel>> getMovies() {
        return watchListRepository.getMovies();
    }

    // Call the method getWatchList in WatchListRepository
    public void getWatchList(String accountId, String sessionId){
        watchListRepository.getWatchList(accountId,sessionId);
    }
}

File: app/src/main/java/me/davidpcosta/tmdb/ui/main/watchlist/WatchlistViewModel.kt
package me.davidpcosta.tmdb.ui.main.watchlist

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import me.davidpcosta.tmdb.data.repository.WatchlistRepository
import me.davidpcosta.tmdb.data.model.Movie

class WatchlistViewModel(private val watchlistRepository: WatchlistRepository) : ViewModel() {

    val movies: LiveData<List<Movie>> = MutableLiveData()

    fun fetchWatchlist(accountId: Long, sessionId: String) {
        watchlistRepository.watchlist(accountId, sessionId).subscribe {
            movies as MutableLiveData
            movies.value = it.results
        }
    }
}